target/linux/rockchip/modules.mk

```
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2024 OpenWrt.org
+
+define KernelPackage/crypto-hw-rockchip
+    SUBMENU:=$(CRYPTO_MENU)
+    TITLE:=Rockchip Cryptographic Engine driver
+    DEPENDS:= \
+	@LINUX_6_6 \
+	@TARGET_rockchip \
+	+kmod-crypto-cbc \
+	+kmod-crypto-des \
+	+kmod-crypto-ecb \
+	+kmod-crypto-hash \
+	+kmod-crypto-kpp \
+	+kmod-crypto-md5 \
+	+kmod-crypto-rsa \
+	+kmod-crypto-sha1 \
+	+kmod-crypto-sha256
+    KCONFIG:= \
+	CONFIG_CRYPTO_AES \
+	CONFIG_CRYPTO_ENGINE \
+	CONFIG_CRYPTO_HW=y \
+	CONFIG_CRYPTO_DEV_ROCKCHIP \
+	CONFIG_CRYPTO_DEV_ROCKCHIP_DEBUG=n \
+	CONFIG_CRYPTO_LIB_DES \
+	CRYPTO_SKCIPHER
+    FILES:= \
+	$(LINUX_DIR)/crypto/crypto_engine.ko \
+	$(LINUX_DIR)/drivers/crypto/rockchip/rk_crypto.ko
+    AUTOLOAD:=$(call AutoLoad,09,crypto_engine rk_crypto)
+    $(call AddDepends/crypto)
+endef
```

```suggestion
define KernelPackage/crypto-hw-rockchip
  SUBMENU:=$(CRYPTO_MENU)
  TITLE:=Rockchip Cryptographic Engine driver
  DEPENDS:= \
	@LINUX_6_6 \
	@TARGET_rockchip \
	+kmod-crypto-cbc \
	+kmod-crypto-des \
	+kmod-crypto-ecb \
	+kmod-crypto-hash \
	+kmod-crypto-kpp \
	+kmod-crypto-md5 \
	+kmod-crypto-rsa \
	+kmod-crypto-sha1 \
	+kmod-crypto-sha256
  KCONFIG:= \
	CONFIG_CRYPTO_AES \
	CONFIG_CRYPTO_DEV_ROCKCHIP \
	CONFIG_CRYPTO_DEV_ROCKCHIP_DEBUG=n \
	CONFIG_CRYPTO_ENGINE \
	CONFIG_CRYPTO_HW=y \
	CONFIG_CRYPTO_LIB_DES \
	CONFIG_CRYPTO_SKCIPHER
  FILES:= \
	$(LINUX_DIR)/crypto/crypto_engine.ko \
	$(LINUX_DIR)/drivers/crypto/rockchip/rk_crypto.ko
  AUTOLOAD:=$(call AutoLoad,09,crypto_engine rk_crypto)
endef
```

use two spaces for indentation, sort Kconfig alphabetically
`AddDepends/crypto` can be omitted here

---

I will open a separate PR for the rockchip crypto driver with V2 that has support  for rk3568.